src/Microsoft.ML.Transforms/StatefulCustomMappingTransformer.cs

```diff
@@ -0,0 +1,398 @@
+        /// an input data with schema <paramref name="inputSchema"/>.
+        /// </summary>
+        public DataViewSchema GetOutputSchema(DataViewSchema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            var rowToRow = new RowToRowMapper(_host, this, new EmptyDataView(_host, inputSchema));
+            return rowToRow.OutputSchema;
+        }
+
+        /// <summary>
+        /// Take the data in, make transformations, output the data.
+        /// Note that <see cref="IDataView"/>'s are lazy, so no actual transformations happen here, just schema validation.
+        /// </summary>
+        public IDataView Transform(IDataView input)
+        {
+            _host.CheckValue(input, nameof(input));
+            return new RowToRowMapper(_host, this, input);
+        }
+
+        /// <summary>
+        /// Constructs a row-to-row mapper based on an input schema. If <see cref="ITransformer.IsRowToRowMapper"/>
+        /// is <c>false</c>, then an exception is thrown. If the <paramref name="inputSchema"/> is in any way
+        /// unsuitable for constructing the mapper, an exception is likewise thrown.
+        /// </summary>
+        IRowToRowMapper ITransformer.GetRowToRowMapper(DataViewSchema inputSchema)
+        {
+            _host.CheckValue(inputSchema, nameof(inputSchema));
+            return new RowToRowMapper(_host, this, new EmptyDataView(_host, inputSchema));
+        }
+
+        private sealed class RowToRowMapper : RowToRowMapperTransformBase
+        {
+            private readonly StatefulCustomMappingTransformer<TSrc, TDst, TState> _parent;
+            private readonly ColumnBindings _bindings;
+            private readonly TypedCursorable<TSrc> _typedSrc;
+
+            public override DataViewSchema OutputSchema => _bindings.Schema;
+
+            public RowToRowMapper(IHostEnvironment env, StatefulCustomMappingTransformer<TSrc, TDst, TState> parent, IDataView input)
+                : base(env, "StatefulCustom", input)
+            {
+                Host.CheckValue(parent, nameof(parent));
+
+                _parent = parent;
+
+                var dstRow = new DataViewConstructionUtils.InputRow<TDst>(Host, _parent.AddedSchema);
+                // All the output columns of dstRow are our outputs.
+                var cols = Enumerable.Range(0, dstRow.Schema.Count).Select(x => new DataViewSchema.DetachedColumn(dstRow.Schema[x])).ToArray();
+
+                _bindings = new ColumnBindings(input.Schema, cols);
+
+                _typedSrc = TypedCursorable<TSrc>.Create(Host, input, false, _parent.InputSchemaDefinition);
+            }
+
+            public override DataViewRowCursor[] GetRowCursorSet(IEnumerable<DataViewSchema.Column> columnsNeeded, int n, Random rand = null)
+            {
+                return new[] { GetRowCursorCore(columnsNeeded, rand) };
+            }
```

Comment:
Just curious. So I'm guessing that by simply doing this we're avoiding this Transform to use multiple threads, right? is there anything else that needs to be changed to stop this transform from using multiple threads?